### ISSUE TITLE
Auth Error

### ISSUE THREAD

**Issue by daviddever** [linuxwhatelse/plugin.audio.linuxwhatelse.gmusic#18]
I think Google may have just made some sort of change on their End, as this plugin has been working great up until a little bit ago today when after finishing a song it was not able to continue with the album playlist.

The play client on my android phone is working through my network, and I tried clearing the cache and rerunning through the setup but got the same results. In the kodi.log I see a lot of errors similar to this, with `401 Client Error: Unauthorized for url` in the json response.

```
13:34:19.921 T:139922753963776   ERROR: [Google Play Music] Traceback (most recent call last):
                                              File "/home/kodi/.kodi/addons/plugin.audio.linuxwhatelse.gmusic/addon/gmusic_wrapper.py", line 48, in _make_call
                                                return super(GMusic, self)._make_call(protocol, *args, **kwargs)
                                              File "/home/kodi/.kodi/addons/script.module.gmusicapi/lib/gmusicapi/clients/shared.py", line 84, in _make_call
                                                return protocol.perform(self.session, self.validate, *args, **kwargs)
                                              File "/home/kodi/.kodi/addons/script.module.gmusicapi/lib/gmusicapi/protocol/shared.py", line 226, in perform
                                                raise CallFailure(err_msg, call_name)
                                            CallFailure: ListListenNowSituations: 401 Client Error: Unauthorized for url: https://mclients.googleapis.com/sj/v2.5/listennow/situations?dv=0&tier=fr&alt=json&hl=en_US
                                            (requests kwargs: {u'url': u'https://mclients.googleapis.com/sj/v2.5/listennow/situations', u'headers': {u'Content-Type': u'application/json', u'Authorization': u'<omitted>'}, u'data': '{"requestSignals": {"timeZoneOffsetSecs": -28800}}', u'method': u'POST', u'params': {u'dv': 0, u'tier': u'fr', u'alt': u'json', u'hl': 'en_US'}})
                                            (response was: u'{"error":{"errors":[{"domain":"global","reason":"authError","message":"Invalid Credentials","locationType":"header","location":"Authorization"}],"code":401,"message":"Invalid Credentials"}}')
```

Thanks for taking look!

**Comment by tadly** [comment]
This is actually somewhat common and will go away on its own.
Up until now I have no clue why this is happening and what to do about it.

Just something we gotta deal with as part of an unofficial client I guess.

I do have one idea which I might take a look at but for now you have to wait and try every now and again.

**Comment by tadly** [comment]
I just had a quick look and got unauthorized responses as well.
After re-running the setup it was work again just fine.

Either way, the idea I was referring to was that I actually test for a valid auth token only once every 3 hours and just expect it to work within those 3 hours.
I did this to increase execution speed as validating the login would take at least one second (which will add up over time).

I have reduced this time-frame to _every hour_ now (495ba4251215e6172c6333e5fd9c665436663859) so lets see if that helps preventing future issues.

**Edit:**
Yes, this is already in the repository as version 1.8.1

**Comment by daviddever** [comment]
Hey thanks for explaining that. I have loaded the new version, and tried logging in a couple of times and ran into some issues but I am guessing these are on Google's end. Last time I tried I did get the notice from Google asking if it was an authorized login so I am guessing next time I try I will probably have better luck.

Thanks again for taking a look and for all the work on this addon.

**Comment by tadly** [comment]
Hm. If it's still not working something else might be wrong which could also by the add-ons fault.
Best thing would probably be to delete the addons `settings.xml` and than try again.
The file is located here `.kodi/userdata/addon_data/plugin.audio.linuxwhatelse.gmusic/settings.xml`

If that doesn't help either, than we gotta dig deeper :)

**Edit:**
Lets keep the issue open until we got that resolved :)

**Comment by daviddever** [comment]
Ok so I got it working again, I was trying to space out my setup attempts so I didn't trigger some kind of timeout lock on Google's side. 

Here is what I figured out, and apologies if this is what the instructions exactly said to do but I didn't read them carefully.

When I go through the setup, ( I have 2 factor authentication on my account and an android device) if I enter the `Android Device Id` from the Device ID App the login fails but in the log I see an error saying I have an invalid ID and then gives me a list of Device ID's and the first one matches what the app shows as `Google Service Framework`.

```
 Error Type: <class 'gmusicapi.exceptions.InvalidDeviceId'>
                                            Error Contents: Invalid device_id <My Device ID>.Your valid device IDs are:
                                            * <String that matches my Google Service Framework Number>                                          
                                            Traceback (most recent call last):
```

When I use that number I was able to login and everything is working again. Apologies if that is what the instructions said to do and I was just going through things to fast.
